utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -112,6 +118,35 @@ fun File.unpack7Zip(targetDirectory: File) {
  */
 fun File.unpackZip(targetDirectory: File) = ZipFile(this).unpack(targetDirectory)
 
+/**
+ * A list with file names that are expected to be contained in a Debian package archive file.
+ */
+private val DEBIAN_ARCHIVE_CONTENT = listOf("data.tar.xz", "control.tar.xz")
+
+/**
+ * Unpack the [File] assuming it is a Debian archive. A Debian archive is an ar archive that contains itself two tar
+ * files with metadata and the actual content of the package. The top-level archive is deflated into a temporary
+ * directory. Then the tar files are extracted to the provided [targetDirectory].
+ */
+fun File.unpackDeb(targetDirectory: File) {
+    val tempDir = createOrtTempDir("unpackDeb")
+
+    try {
+        ArArchiveInputStream(inputStream()).unpack(
+            tempDir,
+            { entry -> entry.isDirectory || File(entry.name).isAbsolute },
+            { entry -> (entry as ArArchiveEntry).mode }
+        )
+
+        DEBIAN_ARCHIVE_CONTENT.forEach { name ->
+            val file = tempDir.resolve(name)
+            file.unpack(targetDirectory)
+        }
```

Comment:
I'm surprised that this is still needed, as I was reading [this comment](https://stackoverflow.com/questions/7432223/open-debian-package-with-java/7553295#comment9037704_7432828) so that Commons Compress would do it automatically... but then again, it probably can't, as it's seeing the `.deb` only as an `.ar`, not knowing about the "nested tar convention".

Reply:
Yes, Commons Compress only extracts the two archive files from the original package.